crates/stages/src/error.rs

```diff
@@ -7,17 +7,28 @@ use reth_primitives::SealedHeader;
 use thiserror::Error;
 use tokio::sync::mpsc::error::SendError;
 
+/// Represents the specific error type within a block error.
+#[derive(Error, Debug)]
+pub enum BlockErrorKind {
+    /// The block encountered a validation error.
+    #[error("Validation error: {0}")]
+    Validation(#[source] consensus::ConsensusError),
+    /// The block encountered an execution error.
+    #[error("Execution error: {0}")]
+    Execution(#[source] executor::BlockExecutionError),
+}
+
 /// A stage execution error.
 #[derive(Error, Debug)]
 pub enum StageError {
-    /// The stage encountered a state validation error.
-    #[error("Stage encountered a validation error in block {number}: {error}.", number = block.number)]
-    Validation {
-        /// The block that failed validation.
+    /// The stage encountered an error related to a block.
+    #[error("Stage encountered a block error in block {number}: {error}.", number = block.number)]
+    BlockError {
```

Comment:
Should it be just `Block` similar to other variants?

Reply:
@shekhirin Yes correct, just changed that